v-next/core/src/type-guards.ts

```diff
@@ -0,0 +1,70 @@
+import type { TaskDefinition } from "./types/tasks.js";
+
+import {
+  ArgumentType,
+  type OptionDefinition,
+  type PositionalArgumentDefinition,
+} from "./types/arguments.js";
+import { TaskDefinitionType } from "./types/tasks.js";
+
+function isValidEnumValue(
+  theEnum: Record<string, string>,
```

Comment:
Double check: do we also want to allow numbers? Record<string, string|number> ?

Reply:
This is the same function we're using in Ignition, I just copied it over. My assumption would be that we probably wouldn't use numbers here as the only time this would be used is to verify one of our internal enums that is always (as far as I know) initialized with strings.